adapters/repos/db/index.go

```diff
@@ -627,13 +630,18 @@ func (i *Index) putObject(ctx context.Context, object *storobj.Object,
 	}
 
 	// no replication, remote shard (or local not yet inited)
-	shard := i.localShard(shardName)
+	shard, release, err := i.getLocalShardNoShutdown(shardName)
+	if err != nil {
+		return err
+	}
+
 	if shard == nil {
 		if err := i.remote.PutObject(ctx, shardName, object, schemaVersion); err != nil {
 			return fmt.Errorf("put remote object: shard=%q: %w", shardName, err)
 		}
 		return nil
 	}
+	defer release()
 
 	// no replication, local shard
 	i.backupMutex.RLock()
```

Comment:
just wondering if `backupMutex` could be used to prevent shutdown as it's being already acquired during insertions...

Reply:
mutex is acquired for the entire class/index being backed up, so its purpose is a bit different.
but I can agree, that we have to many locks at the moment and need to reduce that number soon as code is getting harder to maintain